### PR TITLE
updated versioning to semantic versioning 

### PR DESCRIPTION
**Pull request by Hadis-Fard**
related to #282

### CODE REVIEW

File: source/shared/version.h [hunk 1]
```diff
@@ -24,13 +24,24 @@
...
+#define SEMVER_BUILDMETA
+
+#if SQLVERSION_BUILD > 0
+#undef SEMVER_BUILDMETA
+#define SEMVER_BUILDMETA "+" STRINGIFY( SQLVERSION_BUILD )
+#endif
+
+#define VER_FILEVERSION_STR     STRINGIFY( SQLVERSION_MAJOR ) "." STRINGIFY( SQLVERSION_MINOR ) "." STRINGIFY( SQLVERSION_PATCH ) SEMVER_PRERELEASE SEMVER_BUILDMETA
+#define _FILEVERSION            SQLVERSION_MAJOR,SQLVERSION_MINOR,SQLVERSION_PATCH,SQLVERSION_BUILD
+
```
Comment (yitam) [inline source/shared/version.h]: so when SQLVERSION_BUILD is 1, VER_FILEVERSION_STR is 4.1.7-preview+1?

Comment (Hadis-Fard) [inline source/shared/version.h]: that's true and when it is 0, `VER_FILEVERSION_STR`  is `4.1.7-preview`, and build number is not counted in precedence order.

Comment (yitam) [inline source/shared/version.h]: How about Product name? Both drivers have the same name but file descriptions are different?

Comment (Hadis-Fard) [inline source/shared/version.h]: will remain untouched

**Pull request merged**